hedera-node/src/test/java/com/hedera/services/txns/consensus/SubmitMessageTransitionLogicTest.java

```diff
@@ -105,34 +112,33 @@ void hasCorrectApplicability() {
 	void followsHappyPath() {
 		// given:
 		givenValidTransactionContext();
-
+		final var happyTopic = new Topic(Id.fromGrpcTopic(asTopic(TOPIC_ID)));
+		given(topicStore.loadTopic(any())).willReturn(happyTopic);
 		// when:
 		subject.doStateTransition();
 
 		// then:
-		var topic = topics.get(EntityNum.fromTopicId(asTopic(TOPIC_ID)));
-		assertNotNull(topic);
-		assertEquals(1L, topic.getSequenceNumber()); // Starts at 0.
+		assertEquals(1L, happyTopic.getSequenceNumber()); // Starts at 0.
 
 		// Hash depends on prior state of topic (default topic object has 0s for runningHash and 0L for seqNum),
 		// consensus timestamp, message.
 		assertEquals("c44860f057eca2ea865821f5211420afe231dc2a485c277405d14f8421bb97f4a34ddd53db84bcf064045d10e7fca822",
-				CommonUtils.hex(topic.getRunningHash()));
+				CommonUtils.hex(happyTopic.getRunningHash()));
 
-		verify(transactionContext).setStatus(SUCCESS);
+//		verify(transactionContext).setStatus(SUCCESS);
 	}
 
 	@Test
 	void failsWithEmptyMessage() {
 		// given:
 		givenTransactionContextNoMessage();
-
+		
 		// when:
-		subject.doStateTransition();
+		assertFailsWith(() -> subject.doStateTransition(), INVALID_TOPIC_MESSAGE);
 
 		// then:
 		assertUnchangedTopics();
-		verify(transactionContext).setStatus(INVALID_TOPIC_MESSAGE);
+//		verify(transactionContext).setStatus(INVALID_TOPIC_MESSAGE);
```

Comment:
nit: Should this be commented?

Reply:
Yes, for both of your comments.
 It's no longer in the scope of the transition logic, but `TransactionRecordService`  tracks changes in a manner similar to token, e.g. `includeChangesToToken`.
 The record service is currently injected in the `TopicStore`